### ISSUE TITLE
Different link preloads for static rendering 

### ISSUE THREAD

**Issue by PH4NTOMiki** [sveltejs/kit#4158]
### Describe the bug

Currently `adapter-static` renders same HTML output(.html files) for all `spa: true` / `fallback` mode routes.

What should happen is different link preloads for every HTML page(file) so loading happens quicker.

Kinda related: #4157

### Reproduction

.

### System Info

```shell
latest SvelteKit
```

### Severity

serious, but I can work around it

**Comment by Conduitry** [comment]
I don't think this makes sense. How do you have multiple SPA fallbacks with static files?

**Comment by PH4NTOMiki** [comment]
I don't, there is just one(`index.html`), but other routes get outputed if there is no dynamic parts/slugs in them.

**Comment by Rich-Harris** [comment]
What link preloads are we talking about?

**Comment by PH4NTOMiki** [comment]
<link rel=modulepreload>

**Comment by Rich-Harris** [comment]
Right, but which modules are we preloading? We can't know which modules a given page depends on without server-rendering them, but the whole reason to use SPA+fallback is that you _can't_ server-render those pages.

**Comment by PH4NTOMiki** [comment]
Yeah, but I'm not specifically talking about fallback page(`200.html` for ex.) but other routes that are not server rendered(`{ssr:false}`) but create their own html files(`books.html` in my case is one of them) needs resources `/_app/pages/__layout.svelte-xxxxxxxx.js` (every page including fallback should include Layout preload) and `/_app/pages/books/index.svelte-xxxxxxxx.js` and we know those things in advance so we should include them when prerendering because it delays rendering(slightly) to have to parse `start.js` to find what resources it needs to render the current page. 

This is related to issue #4157 I also created where I'm requesting to prerender even the parameterized routes when `{ssr:false}` and `adapter-static` is being used so I can manually create rewrite file for my specific hosting service for routes, essentially my request in that issue is if there is a route for `/users/[user_id].svelte` and `ssr` is `false` to prerender/output it as `/users/_slug_id_.html` or something similar, and in that case we do link preloads for them too, and that would make fallback file(at least for my case, and I believe for some others too) unnecessary.

Does that makes sense?

**Comment by PH4NTOMiki** [comment]
@Rich-Harris I could maybe try to do PR for this, but I don't know much about Svelte/Vite compilation

**Comment by teemingc** [comment]
I _think_ this is resolved by https://github.com/sveltejs/kit/pull/15718
